### PR TITLE
Enhancement/add another bottom label to the cell

### PR DESCRIPTION
**Pull request by maxxx777**
What does this implement/fix? Explain your changes.
---------------------------------------------------
It adds cell bottom label into message content cell.

Does this close any currently open issues?
------------------------------------------
#769 

Any relevant logs, error output, etc?
-------------------------------------
the screenshot of the result:
simulator screen shot - iphone x - 2018-10-22 at 00 29 12

Any other comments?
-------------------
-

Where has this been tested?
---------------------------
**Devices/Simulators:** iPhone X (Simulator)

**iOS Version:** 12.0

**Swift Version:** 4.2

**MessageKit Version:** 2.0.0-beta.1

### CODE REVIEW

File: Example/Sources/View Controllers/BasicExampleViewController.swift [hunk 1]
```diff
@@ -34,6 +34,11 @@ final class BasicExampleViewController: ChatViewController {
         
         messagesCollectionView.messagesLayoutDelegate = self
         messagesCollectionView.messagesDisplayDelegate = self
+        
+        let layout = messagesCollectionView.collectionViewLayout as? MessagesCollectionViewFlowLayout
+        
+        layout?.setMessageIncomingAvatarPosition(AvatarPosition(vertical: .messageBottom))
```
Comment (nathantannar4) [inline Example/Sources/View Controllers/BasicExampleViewController.swift]: Isn't this the default position?

File: Sources/Layout/MessageSizeCalculator.swift [hunk 2]
```diff
@@ -181,8 +188,23 @@ open class MessageSizeCalculator: CellSizeCalculator {
...
+    }
+    
+    open func cellBottomLabelAlignment(for message: MessageType) -> LabelAlignment {
+        let dataSource = messagesLayout.messagesDataSource
+        let isFromCurrentSender = dataSource.isFromCurrentSender(message: message)
+        return isFromCurrentSender ? outgoingCellBottomLabelAlignment : incomingCellBottomLabelAlignment
+    }
 
-    // MARK: - Bottom Label
+    // MARK: - Bottom message Label
```
Comment (nathantannar4) [inline Sources/Layout/MessageSizeCalculator.swift]: Capitalize `message`

**Review by nathantannar4 — CHANGES_REQUESTED** [review]
Thanks so much for taking this on!! 🔥 💯

File: Example/Sources/View Controllers/BasicExampleViewController.swift (hunk 1 above)
Comment (maxxx777) [reply Example/Sources/View Controllers/BasicExampleViewController.swift]: Yes, this is not the default position. I reverted back the change not to confuse users and not to complicate the example project.

File: Sources/Layout/MessageSizeCalculator.swift (hunk 2 above)
Comment (maxxx777) [reply Sources/Layout/MessageSizeCalculator.swift]: Fixed

**Comment by nathantannar4** [comment]
LGTM, @SD10

File: Sources/Views/Cells/MessageContentCell.swift [hunk 3]
```diff
@@ -81,6 +89,7 @@ open class MessageContentCell: MessageCollectionViewCell {
     open func setupSubviews() {
         contentView.addSubview(accessoryView)
         contentView.addSubview(cellTopLabel)
+        contentView.addSubview(cellBottomLabel)
```
Comment (SD10) [inline Sources/Views/Cells/MessageContentCell.swift]: Nit: Can you add this subview _after_ the `messageBottomLabel`?

File: Sources/Views/Cells/MessageContentCell.swift [hunk 4]
```diff
@@ -253,6 +268,18 @@ open class MessageContentCell: MessageCollectionViewCell {
         cellTopLabel.frame = CGRect(origin: .zero, size: attributes.cellTopLabelSize)
     }
     
+    /// Positions the cell's bottom label.
+    /// - attributes: The `MessagesCollectionViewLayoutAttributes` for the cell.
+    open func layoutCellBottomLabel(with attributes: MessagesCollectionViewLayoutAttributes) {
+        cellBottomLabel.textAlignment = attributes.cellBottomLabelAlignment.textAlignment
+        cellBottomLabel.textInsets = attributes.cellBottomLabelAlignment.textInsets
+        
+        let y = messageContainerView.frame.maxY + attributes.messageContainerPadding.bottom + attributes.messageBottomLabelSize.height
```
Comment (SD10) [inline Sources/Views/Cells/MessageContentCell.swift]: For calculating the Y position, can we make it so the `cellBottomLabel` always starts after the `messageBottomLabel` max Y?

File: Sources/Views/Cells/MessageContentCell.swift [hunk 5]
```diff
@@ -265,9 +292,9 @@ open class MessageContentCell: MessageCollectionViewCell {
         messageTopLabel.frame = CGRect(origin: origin, size: attributes.messageTopLabelSize)
     }
 
-    /// Positions the cell's bottom label.
+    /// Positions the message bubble's bottom label.
     /// - attributes: The `MessagesCollectionViewLayoutAttributes` for the cell.
-    open func layoutBottomLabel(with attributes: MessagesCollectionViewLayoutAttributes) {
+    open func layoutMessageBottomLabel(with attributes: MessagesCollectionViewLayoutAttributes) {
```
Comment (SD10) [inline Sources/Views/Cells/MessageContentCell.swift:297]: This is an API breaking change. Let's make sure to add a CHANGELOG entry for this. Please refer to the other breaking change entries and follow that format.

File: Sources/Views/Cells/MessageContentCell.swift [hunk 6]
```diff
@@ -46,6 +46,14 @@ open class MessageContentCell: MessageCollectionViewCell {
         return label
     }()
     
+    /// The bottom label of the cell.
+    open var cellBottomLabel: InsetLabel = {
```
Comment (SD10) [inline Sources/Views/Cells/MessageContentCell.swift:50]: Let's add a CHANGELOG entry for this new feature 👍

**Review by SD10 — CHANGES_REQUESTED** [review]
Hey @maxxx777, thanks for taking this on 💯 Looks really solid to me. Just please take a look at my suggestions

**Comment by SD10** [comment]
This has to go into 3.0 because it's API breaking and tbh a pretty big layout change. Let's also start using the API breaking label

File: Sources/Views/Cells/MessageContentCell.swift (hunk 4 above)
Comment (maxxx777) [reply Sources/Views/Cells/MessageContentCell.swift]: I think it's better if the calculation of the layout of one cell does not depend on the layout of another cell. 
For example, the layout of `messageBottomLabel` is not calculated yet. if we gonna use it to calculate the layout of `cellBottomLabel`, we get the wrong layout of `cellBottomLabel`.
We have to agree either to use a certain order of the calculation of labels' layouts in `apply(_:)` function, or depend only on `messageContainerView`s layout in the calculations of labels' layouts.

Comment (SD10) [reply Sources/Views/Cells/MessageContentCell.swift]: Fwiw, the rest of the layout methods rely on a certain order of method calls. I think making them dependent is a more reliable and easier way to do the layout

Comment (maxxx777) [reply Sources/Views/Cells/MessageContentCell.swift]: Ok, did the fixes.

**Review by SD10 — APPROVED** [review]
Awesome work here @maxxx777. Sorry for the slow review time. Thanks so much for this 💯

**Pull request merged**